calico_containers/pycalico/datastore.py

```diff
@@ -291,9 +292,25 @@ def ensure_global_config(self):
         # Disable felix status reporting, which consumes etcd write bandwidth.
         self._write_global_config(FELIX_REPORTING_INTERVAL_PATH, "0")
 
+        # Create our Cluster GUID (if it does not already exist).
+        self._create_cluster_guid(CLUSTER_GUID_PATH)
+
         # We are always ready.
         self.etcd_client.write(CALICO_V_PATH + "/Ready", "true")
 
+    def _create_cluster_guid(self, key):
+        """
+        Create a globally unique cluster GUID.  Write it idempotently into the
```

Comment:
Again, fairly subtle, but I think you could make this comment more accurate by saying 
```
Ensures that etcd contains a GUID for the cluster.
Creates the GUID (safely with CaS) if it doesn't exist.
```